R/xml_serialize.R

```diff
@@ -64,9 +64,15 @@ xml_unserialize <- function(connection, ...) {
     # Select only the root
     res <- xml_find_first(x, "/node()")
   } else if (inherits(object, "xml_serialized_document")) {
-    res <- read_xml(unclass(object), ...)
+    read_xml_int <- function(object, as_html = FALSE, ...) {
+      if (missing(as_html)) {
+        as_html <- identical(attr(object, "doc_type", exact = TRUE), "html")
+      }
+      read_xml(unclass(object), as_html = as_html, ...)
+    }
+    res <- read_xml_int(unclass(object), ...)
   } else {
-    abort("Not a serialized xml2 object")
+    stop("Not a serialized xml2 object", call. = FALSE)
```

Comment:
Is there a reason for this change?

Reply:
No, I didn't touch that, and I also wondered about it. Maybe it sneaked in via some merge(?).  Let me, and I'll revert it back to `abort()` .

Reply:
Thanks!

Reply:
Fixed